### PR TITLE
fix(compatibility): remove public key and upgrade nuxt

### PR DESCRIPTION
**Pull request by larbish**
Prevent breaking change in v3.5 of Nuxt: https://github.com/nuxt/nuxt/pull/20082 and avoid having those warning using the v3.4:
Screenshot 2023-04-12 at 16 30 23

### CODE REVIEW

File: src/module.ts [hunk 1]
```diff
@@ -84,7 +84,7 @@ export default defineNuxtModule<ModuleOptions>({
   },
   setup (options, nuxt) {
     // Default runtimeConfig
-    nuxt.options.runtimeConfig.public.strapi = defu(nuxt.options.runtimeConfig.public.strapi, options)
```
Comment (QozbroQqn) [inline src/module.ts:87]: This line was fine

File: src/module.ts [hunk 2]
```diff
@@ -84,7 +84,7 @@ export default defineNuxtModule<ModuleOptions>({
...
+    nuxt.options.runtimeConfig.strapi = defu(nuxt.options.runtimeConfig.strapi, options)
     nuxt.options.runtimeConfig.strapi = defu(nuxt.options.runtimeConfig.strapi, options)
```
Comment (QozbroQqn) [inline src/module.ts:88]: this line can be remove

File: src/module.ts [hunk 3]
```diff
@@ -106,7 +106,7 @@ export default defineNuxtModule<ModuleOptions>({
       config.optimizeDeps.include.push('qs')
     })
 
-    const adminUrl = joinURL(nuxt.options.runtimeConfig.public.strapi.url, '/admin/')
```
Comment (QozbroQqn) [inline src/module.ts:109]: this line was fine

**Comment by QozbroQqn** [comment]
@larbish I think you misunderstand the warning. With 3.5, we **have to** access public runtime variables via `config.public.x`.

There are a few more places where nuxt throws a warning. useStrapiToken composable for example.

**Comment by larbish** [comment]
Indeed ! I was a bit hasty on this one...

**Pull request closed without merging**